src/api/scan/base_controller.go

```diff
@@ -63,31 +61,33 @@ type configGetter func(cfg string) (string, error)
 // jcGetter is a func template which is used to get the job service client.
 type jcGetter func() cj.Client
 
+// metadataGetter is a func template which is used to get the metadata of the registration
+type metadataGetter func(r *scanner.Registration) (*v1.ScannerAdapterMetadata, error)
+
+// robotGetter is a func template which is used to make robot account for scanner
+type robotGetter func(projectID int64) (*model.Robot, error)
+
 // basicController is default implementation of api.Controller interface
 type basicController struct {
 	// Manage the scan report records
 	manager report.Manager
-	// Scanner controller
-	sc sc.Controller
-	// Robot account controller
-	rc robot.Controller
 	// Job service client
 	jc jcGetter
 	// UUID generator
 	uuid uuidGenerator
 	// Configuration getter func
 	config configGetter
+	// Metadata getter func
+	metadata metadataGetter
+	// Robot account getter func
+	robot robotGetter
 }
 
 // NewController news a scan API controller
 func NewController() Controller {
 	return &basicController{
 		// New report manager
 		manager: report.NewManager(),
-		// Refer to the default scanner controller
-		sc: sc.DefaultController,
-		// Refer to the default robot account controller
-		rc: robot.RobotCtr,
 		// Refer to the default job service client
 		jc: func() cj.Client {
```

Comment:
The reason for this callback is because the initialization of the default jobservice client may be after this scan controller. If we have another better choice, maybe we can remove it too.